### ISSUE TITLE
`$.each`: add support for plain objects

### ISSUE THREAD

**Issue by softwarespot** [fabiospampinato/cash#104]
I was just wandering whether this is a design choice, in that each only works with arrays?

**Comment by simeydotme** [comment]
Yes, while doing new unit tests, I also wondered this...

looking at the current unit tests: https://github.com/kenwheeler/cash/blob/master/test/test.js#L133-L139 it seems to be left out... although so does `$.each( $("<div>") );` ... but that works... so I'm lost, @kenwheeler it'd be great to know if this is an oversight or design decision? :)

**Comment by softwarespot** [comment]
That works as it's an iterable object i.e. array-like

**Comment by shshaw** [comment]
Here's the `each` function I normally use, which is quite similar to Cash's but does support objects. It wouldn't add much weight to cash. Is that something we want to support?

```
function each(obj,callback,thisArg) {
  var length = obj.length,
      likeArray = ( length === 0 || ( length > 0 && (length - 1) in obj ) ),
      i = 0;

  if ( likeArray ) {
    for ( ; i < length; i++ ) {
      if ( callback.call( thisArg || obj[i], obj[ i ], i, obj ) === false ) { break; }
    }
  } else {
    for (i in obj) {
      if ( callback.call( thisArg || obj[i], obj[ i ], i, obj ) === false ) { break; }
    }
  }
  
  return obj;
}
```

**Comment by shshaw** [comment]
Something else I've noticed... `jQuery.fn.each` gives callbacks the arguments in the opposite order that we have: `( Integer index, Element element )` ( index first, element/object second )

In the interest of code portability, should we reverse our arguments as well?

**Comment by leeoniya** [comment]
> In the interest of code portability, should we reverse our arguments as well?

yes, you absolutely should. countless plugins interbally rely on `i´ being arg 0 within jquery's $.fn.each and $.each. this subtle difference manifests itself in unhelpful debug messages rather than "call to undefined method X".

this has proven to be a showstopper for us to cash adoption. we're fine polyfilling missing methods but those that are supported should work like jquery's and support at minimum the most common signatures.

**Comment by fabiospampinato** [comment]
Adding support for objects to `$.each` seems unnecessary to me, but maybe we could also use it internally and shave some bytes? 🤔 If somebody wants to experiment with this I'll be happy to consider a PR that adds very few bytes to the bundle and basically doesn't affect performance.

> In the interest of code portability, should we reverse our arguments as well?

> this has proven to be a showstopper for us to cash adoption. we're fine polyfilling missing methods but those that are supported should work like jquery's and support at minimum the most common signatures.

As of v2.1.2 cash's each/map/filter pass arguments in the same order as jQuery's each/map/filter.

**Comment by fabiospampinato** [comment]
Implemented in #347
